src/Integration.UnitTests/Binding/RoslynProjectBinderTests.cs

```diff
@@ -0,0 +1,213 @@
+            result.Should().BeFalse();
+        }
+
+        [TestMethod]
+        public void IsBound_RulesetFileHasNoDeclarations_False()
+        {
+            var bindingConfiguration = GetBindingConfiguration();
+            var projectMock = GetCSharpProject();
+            var mockRulesetPath = SetupRulesetPath();
+
+            fileSystemMock.Setup(x => x.File.Exists(mockRulesetPath)).Returns(true);
+            
+            ruleSetSerializerMock.Setup(x => x.LoadRuleSet(mockRulesetPath)).Returns(new RuleSet("test"));
+
+            solutionRuleSetsInformationProviderMock
+                .Setup(x => x.GetProjectRuleSetsDeclarations(projectMock))
+                .Returns(Array.Empty<RuleSetDeclaration>());
+
+            var result = testSubject.IsBound(bindingConfiguration, projectMock);
+            result.Should().BeFalse();
+        }
+
+        [TestMethod]
+        public void IsBound_RulesetFileHasOneDeclaration_CantLoadProjectRulesetFile_False()
+        {
+            var bindingConfiguration = GetBindingConfiguration();
+            var projectMock = GetCSharpProject();
+            var mockRulesetPath = SetupRulesetPath();
+
+            fileSystemMock.Setup(x => x.File.Exists(mockRulesetPath)).Returns(true);
+
+            ruleSetSerializerMock.Setup(x => x.LoadRuleSet(mockRulesetPath)).Returns(new RuleSet("test"));
+
+            var ruleSetDeclaration = GetRuleSetDeclaration(projectMock);
+
+            solutionRuleSetsInformationProviderMock
+                .Setup(x => x.GetProjectRuleSetsDeclarations(projectMock))
+                .Returns(new List<RuleSetDeclaration> {ruleSetDeclaration});
+
+            var filePath = "";
+            solutionRuleSetsInformationProviderMock
+                .Setup(x => x.TryGetProjectRuleSetFilePath(projectMock, ruleSetDeclaration, out filePath))
+                .Returns(false);
+
+            var result = testSubject.IsBound(bindingConfiguration, projectMock);
+            result.Should().BeFalse();
+        }
+
+        private RuleSetDeclaration GetRuleSetDeclaration(ProjectMock projectMock)
+        {
+            var mockDeclaration =
+                new RuleSetDeclaration(projectMock, new PropertyMock("name", null), "test path", null);
+            
+            return mockDeclaration;
+        }
+
+        private string SetupRulesetPath()
+        {
+            var mockRulesetPath = "c:\\test.ruleset";
+
+            solutionRuleSetsInformationProviderMock
+                .Setup(x => x.CalculateSolutionSonarQubeRuleSetFilePath("key", Language.CSharp, SonarLintMode.Connected))
+                .Returns(mockRulesetPath);
+
+            return mockRulesetPath;
+        }
+
+        private static BindingConfiguration GetBindingConfiguration()
+        {
+            var bindingConfiguration =
+                new BindingConfiguration(new BoundSonarQubeProject(new Uri("http://a.com"), "key", "name"),
+                    SonarLintMode.Connected, "dummy directory");
```

Comment:
It's not obvious which bits of data in the binding are meaningful and which are junk data that can be safely ignored when reading the tests. Using values like `dummy directory` helps, or `any` / `junk`.

As it happens, it looks like the value used for the project `key` does matter since the same value has to be used for in `SetupRulesPath`. The connection isn't obvious which makes the tests fragile.
The connection could be made more obvious by using a constant, or by passing the `BindingConfiguration` to `SetupRulesPath`.